docs/migration_guide.md

```diff
@@ -0,0 +1,105 @@
+|`bincode::deserialize_from_custom(BincodeRead)`|`bincode::serde::decode_from_reader(Reader, Configuration)`|
+|||
+|`bincode::serialize(T)`|`bincode::serde::encode_to_vec(T, Configuration)`<br />`bincode::serde::encode_into_slice(t: T, &mut [u8], Configuration)`|
+|`bincode::serialize_into(std::io::Write, T)`|`bincode::serde::encode_into_std_write(T, std::io::Write, Configuration)`|
+|`bincode::serialized_size(T)`|Currently not implemented|
+
+## Migrating to `bincode-derive`
+
+`bincode-derive` is enabled by default. If you're using `default-features = false`, make sure to add `features = ["derive"]` to your `Cargo.toml`.
+
+```toml,ignore
+[dependencies]
+bincode = "2.0.0-beta"
+
+# If you need `no_std` with `alloc`:
+# bincode = { version = "2.0.0-beta", default-features = false, features = ["derive", "alloc"] }
+
+# If you need `no_std` and no `alloc`:
+# bincode = { version = "2.0.0-beta", default-features = false, features = ["derive"] }
+```
+
+Replace or add the following attributes. You are able to use both `serde-derive` and `bincode-derive` side-by-side.
+
+|serde-derive|bincode-derive|
+|---|---|
+|`#[derive(serde::Serialize)]`|`#[derive(bincode::Encode)]`|
+|`#[derive(serde::Deserialize)]`|`#[derive(bincode::Decode)]`|
+
+**note:** To implement these traits manually, see the documentation of [Encode](https://docs.rs/bincode/2.0.0-beta/bincode/enc/trait.Encode.html) and [Decode](https://docs.rs/bincode/2.0.0-beta/bincode/de/trait.Decode.html).
+
+**note:** For more information on using `bincode-derive` with external libraries, see [below](#bincode-derive-and-libraries).
+
+Then replace the following functions:
+
+|Bincode 1|Bincode 2|
+|--|--|
+|`bincode::deserialize(&[u8])`|`bincode::decode_from_slice(&bytes, Configuration)`<br />`bincode::decode_borrowed_from_slice(&[u8], Configuration)`|
+|`bincode::deserialize_from(std::io::Read)`|`bincode::decode_from_std_read(std::io::Read, Configuration)`|
+|`bincode::deserialize_from_custom(BincodeRead)`|`bincode::decode_from_reader(Reader, Configuration)`|
+|||
+|`bincode::serialize(T)`|`bincode::encode_to_vec(T, Configuration)`<br />`bincode::encode_into_slice(t: T, &mut [u8], Configuration)`|
+|`bincode::serialize_into(std::io::Write, T)`|`bincode::encode_into_std_write(T, std::io::Write, Configuration)`|
+|`bincode::serialized_size(T)`|Currently not implemented|
+
+
+### Bincode derive and libraries
+
+Currently not many libraries support the traits `Encode` and `Decode`. There are a couple of options if you want to use `#[derive(bincode::Encode, bincode::Decode)]`:
+- Enable the `serde` feature and add a `#[bincode(with_serde)]` above each field that implements `serde` but not `Encode/Decode`
+- Enable the `serde` feature and wrap your field in [bincode::serde::Compat](https://docs.rs/bincode/2.0.0-beta/bincode/serde/struct.Compat.html) or [bincode::serde::BorrowCompat](https://docs.rs/bincode/2.0.0-beta/bincode/serde/struct.BorrowCompat.html)
+- Make a pull request to the library:
```

Comment:
Alternatively they could make a PR to our library to have it as an optional feature right?

Reply:
I'd rather keep the bincode repo as clean as possible. Maybe we can make a bunch of interop crates, e.g. `bincode-uuid` or something? Not sure